coverage_comment/wiki.py

```diff
@@ -67,6 +69,27 @@ def get_file_contents(
         response.raise_for_status()
         return response.text
     except httpx.HTTPError:
+        # One possible reason is that the repository (and hence wiki) is private,
+        # requiring us to clone the repo and get file contents that way
+        if git and github_token:
+            log.debug(
+                "Exception while getting previous coverage data, attempting to download "
+                "the file directly from the git repository (assuming the wiki is private)"
+            )
+            with tempfile.TemporaryDirectory() as dir_path:
+                dir = pathlib.Path(dir_path)
+                git.cwd = dir
+
+                git.clone(
+                    f"https://x-access-token:{github_token}@github.com/{repository}.wiki.git",
+                    ".",
+                )
+
+                try:
+                    return (dir / filename).read_text()
```

Comment:
This is mostly copy-pasted from above. D you think we could refactor that ? For example as a `@contextlib.contextmanager` that would clone the repo and `yield dir` and you'd do `with cloned_wiki(...) as dir:` ?

Also, I believe there is a good coverage currently, so you may want to add some additional tests ?

Reply:
I have added tests - I can't trigger the tests to run and I haven't set up local testing so can't see whether it will pass.